components/timezone/src/zoned_datetime.rs

```diff
@@ -4,35 +4,108 @@
 
 use icu_calendar::{AsCalendar, Date, Iso, Time};
 
-use crate::CustomTimeZone;
+use crate::{FormattableTimeZone, TimeZone};
+use crate::{MetazoneCalculator, ZoneOffsetCalculator};
 
 /// A date and time local to a specified custom time zone.
 #[derive(Debug)]
 #[allow(clippy::exhaustive_structs)] // this type is stable
-pub struct CustomZonedDateTime<A: AsCalendar> {
+pub struct ZonedDateTime<A: AsCalendar> {
     /// The date, local to the time zone
     pub date: Date<A>,
     /// The time, local to the time zone
     pub time: Time,
     /// The time zone
-    pub zone: CustomTimeZone,
+    pub zone: TimeZone,
 }
 
-impl<A: AsCalendar> CustomZonedDateTime<A> {
-    /// Convert the CustomZonedDateTime to an ISO CustomZonedDateTime
+impl<A: AsCalendar> ZonedDateTime<A> {
+    /// Convert the ZonedDateTime to an ISO ZonedDateTime
     #[inline]
-    pub fn to_iso(&self) -> CustomZonedDateTime<Iso> {
-        CustomZonedDateTime {
+    pub fn to_iso(&self) -> ZonedDateTime<Iso> {
+        ZonedDateTime {
             date: self.date.to_iso(),
             time: self.time,
             zone: self.zone,
         }
     }
 
-    /// Convert the CustomZonedDateTime to a CustomZonedDateTime in a different calendar
+    /// Convert the ZonedDateTime to a ZonedDateTime in a different calendar
     #[inline]
-    pub fn to_calendar<A2: AsCalendar>(&self, calendar: A2) -> CustomZonedDateTime<A2> {
-        CustomZonedDateTime {
+    pub fn to_calendar<A2: AsCalendar>(&self, calendar: A2) -> ZonedDateTime<A2> {
+        ZonedDateTime {
+            date: self.date.to_calendar(calendar),
+            time: self.time,
+            zone: self.zone,
+        }
+    }
+
+    /// Convert the [`ZonedDateTime`] to a [`FormattableZonedDateTime`].
+    pub fn to_formattable(
```

Comment:
I like the data model, but I think we need to consider these conversion functions a bit more carefully.

To start, I think the functions could be

```rust
impl<A: AsCalendar> ZonedDateTime<A> {
    #[cfg(feature = "compiled_data")]
    pub fn into_formattable(self) -> FormattableZonedDateTime<A> {
        MetazoneAndZoneOffsetCalculator::new().compute(self)
    }
}

// combined type: discuss below
impl MetazoneAndZoneOffsetCalculator {
    pub fn compute<A: AsCalendar>(&self, zdt: ZonedDateTime<A>) -> FormattableZonedDateTime<A> {
        // use this type's internal data to convert them
    }
}
```

So we have a nice, clean, ergonomic function for the common case, and a still-pretty-clean function that follows all ICU4X conventions for the dynamic data case. In general I think it has been a good call that we generally avoid functions that take dataful objects as arguments, and this lets us do that.

I'm not sure we need these functions on `TimeZone` itself, since you should convert to a `ZonedDateTime` first. However, we do need a `FormattableTimeZone`.